src/Interpreters/AsynchronousInsertQueue.cpp

```diff
@@ -233,7 +233,7 @@ void AsynchronousInsertQueue::preprocessInsertQuery(const ASTPtr & query, const
 
     InterpreterInsertQuery interpreter(query, query_context, query_context->getSettingsRef().insert_allow_materialized_columns);
     auto table = interpreter.getTable(insert_query);
-    auto sample_block = interpreter.getSampleBlock(insert_query, table, table->getInMemoryMetadataPtr());
+    auto sample_block = interpreter.getSampleBlock(insert_query, table, table->getInMemoryMetadataPtr(), query_context);
```

Comment:
`AsynchronousInsertQueue` already checks access in `AsynchronousInsertQueue::processData()`.

Reply:
I believe we don't check access here